stdlib/public/runtime/Errors.cpp

```diff
@@ -22,6 +22,22 @@
 #include <mutex>
 #endif
 
+#if defined(__ELF__)
+# if defined(__FreeBSD__) && !defined(_GNU_SOURCE)
+// In order to access _Unwind_Backtrace() from (the non-LLVM, non-GCC)
+// <unwind.h> in FreeBSD, define _GNU_SOURCE around the include.  Do this
+// before including other headers which may in turn include <unwind.h>, which
+// has a header guard.
+#  define _GNU_SOURCE
+#  define _SHOULD_UNDEFINE_GNU_SOURCE
+# endif
```

Comment:
How terrible would it be to define `_GNU_SOURCE` when building just this source file for FreeBSD?  That would impact any other headers included, but should be generally acceptable I imagine.  That could then sink this into the build system as:

```
set_source_files_properties(Errors.cpp PROPERTIES
  COMPILE_DEFINITIONS $<$<PLATFORM_ID:FreeBSD>:_GNU_SOURCE>)
```